src/CollarTakerNFT.sol

```diff
@@ -106,164 +100,158 @@ contract CollarTakerNFT is ICollarTakerNFT, BaseNFT {
+
+        TakerPosition memory takerPosition = TakerPosition({
+            providerNFT: providerNFT,
+            providerId: providerId,
+            duration: offer.duration,
+            expiration: expiration,
+            startPrice: startPrice,
+            putStrikePrice: putStrikePrice,
+            callStrikePrice: callStrikePrice,
+            takerLocked: takerLocked,
+            providerLocked: providerLocked,
+            // unset until settlement
+            settled: false,
+            withdrawable: 0
+        });
+
+        // increment ID
+        takerId = nextTokenId++;
+        // store position data
+        positions[takerId] = takerPosition;
+        // mint the NFT to the sender, @dev does not use _safeMint to avoid reentrancy
+        _mint(msg.sender, takerId);
+
+        emit PairedPositionOpened(takerId, address(providerNFT), providerId, offerId, takerPosition);
 
         // pull the user side of the locked cash
         cashAsset.safeTransferFrom(msg.sender, address(this), takerLocked);
     }
 
     /// @dev this should be called as soon after expiry as possible, because if the expiry TWAP price becomes
     /// unavailable in the UniV3 oracle, the current price will be used instead of it.
-    /// Both taker and providder should be incentivised to call this method, however it's possible that
+    /// Both taker and provider are incentivised to call this method, however it's possible that
     /// one side is not (e.g., due to being at max loss). For this reason a keeper should be run to
-    /// prevent regular users with gains from neglecting to settle their positions on time.
-    /// @dev To increase the timespan during which the price is available use
-    /// `increaseCardinality` (or the pool's `increaseObservationCardinalityNext`).
+    /// prevent users with gains from not settling their positions on time.
+    /// @dev To increase the timespan during which the historical price is available use
+    /// `oracle.increaseCardinality` (or the pool's `increaseObservationCardinalityNext`).
     function settlePairedPosition(uint takerId) external whenNotPaused {
         TakerPosition storage position = positions[takerId];
 
         require(position.expiration != 0, "position doesn't exist");
         require(block.timestamp >= position.expiration, "not expired");
         require(!position.settled, "already settled");
 
-        // get settlement price. casting is safe since expiration was checked
-        (uint endPrice, bool historical) = oracle.pastPriceWithFallback(uint32(position.expiration));
-
-        (uint withdrawable, int toProvider) = _settlementCalculations(position, endPrice);
+        // settlement price
+        (uint endPrice, bool historical) = oracle.pastPriceWithFallback(position.expiration.toUint32());
+        // settlement amounts
+        (uint takerBalance, int providerDelta) = _settlementCalculations(position, endPrice);
```

Comment:
wasnt toProvider the name preferred for these things in the other pr?

Reply:
Yeah sorry about going back and forth on this. What was confusing still to me in Taker context was the `toProvider` was ambiguous still because it wasn't clear if it's the whole amount the **provider account/actor** gets, or just the delta the **provider contract** gets. So I decided to go back and more clearly name it `providerDelta` in this PR to avoid this confusion.